x-pack/legacy/plugins/siem/public/containers/case/configure/use_configure.tsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useState, useEffect, useCallback } from 'react';
+import { getCaseConfigure, patchCaseConfigure, postCaseConfigure } from './api';
+
+import { useStateToaster, errorToToaster } from '../../../components/toasters';
+import * as i18n from '../translations';
+import { ClosureType } from './types';
+
+interface PersistCaseConfigure {
+  connectorId: string;
+  closureType: ClosureType;
+}
+
+export interface ReturnUseCaseConfigure {
+  loading: boolean;
+  refetchCaseConfigure: () => void;
+  persistCaseConfigure: ({ connectorId, closureType }: PersistCaseConfigure) => Promise<unknown>;
+  persistLoading: boolean;
+}
+
+interface UseCaseConfigure {
+  setConnectorId: (newConnectorId: string) => void;
+  setClosureType: (newClosureType: ClosureType) => void;
+}
+
+export const useCaseConfigure = ({
+  setConnectorId,
+  setClosureType,
+}: UseCaseConfigure): ReturnUseCaseConfigure => {
+  const [, dispatchToaster] = useStateToaster();
+  const [loading, setLoading] = useState(true);
+  const [persistLoading, setPersistLoading] = useState(false);
+  const [version, setVersion] = useState('');
+
+  const refetchCaseConfigure = useCallback(async () => {
+    let didCancel = false;
+    const abortCtrl = new AbortController();
```

Comment:
for `refetchCaseConfigure`, `persistCaseConfigure`, and `updateConnector` what do you think about adopting the closure async/await pattern we [discussed in my PR](https://github.com/elastic/kibana/pull/59616#discussion_r390516617)? https://dev.to/n1ru4l/homebrew-react-hooks-useasynceffect-or-how-to-handle-async-operations-with-useeffect-1fa8. This way in your `handleSubmit` in `siem/public/pages/case/components/configure_cases/index.tsx`, you do not have to await and you can show a loading state

Reply:
I believe that is only applied to `useEffect` and not to `useCallback`.  The `useCallback` returns a memoized version of a function. A function can be `async`. On the other hand, `useEffect` is tight with the lifecycle of the React components and therefore is not advised to await inside `useEffect`.  `refetchCaseConfigure `, `persistCaseConfigure `, and `updateConnector ` need to do async operation so that's why the async. But, if you see inside the `useEffect` the calls to those functions are without `await`.

Reply:
I am missing something? Maybe I do not understand :)

Reply:
Ok, I got it!